src/arch/wasm/CodeGen.zig

```diff
@@ -3869,14 +3871,17 @@ fn airIsNull(func: *CodeGen, inst: Air.Inst.Index, opcode: wasm.Opcode, op_kind:
 /// NOTE: Leaves the result on the stack
 fn isNull(func: *CodeGen, operand: WValue, optional_ty: Type, opcode: wasm.Opcode) InnerError!WValue {
     try func.emitWValue(operand);
+    var buf: Type.Payload.ElemType = undefined;
+    const payload_ty = optional_ty.optionalChild(&buf);
     if (!optional_ty.optionalReprIsPayload()) {
-        var buf: Type.Payload.ElemType = undefined;
-        const payload_ty = optional_ty.optionalChild(&buf);
         // When payload is zero-bits, we can treat operand as a value, rather than
         // a pointer to the stack value
         if (payload_ty.hasRuntimeBitsIgnoreComptime()) {
             try func.addMemArg(.i32_load8_u, .{ .offset = operand.offset(), .alignment = 1 });
         }
+    } else if (payload_ty.isSlice()) {
+        // move the ptr on top of the stack
+        _ = try func.load(operand, Type.usize, 0);
```

Comment:
At line 3873 we already put the address of the operand on the stack. Doing a `load` here like this will mean we have too many operands on the stack and Wasm's stack-validator will fail on the generated code. Instead, this should use the same functionality as above and use `addMemArg(.i32_load, .{.offset = operand.offset(), .alignment = 4}`. This will load the pointer field from the slice, which will then be compared to '0'.